hydra/plugins/sweeper.py

```diff
@@ -51,9 +51,15 @@ def validate_batch_is_legal(self, batch: Sequence[Sequence[str]]) -> None:
         This repeat work the launcher will do, but as the launcher may be performing this in a different
         process/machine it's important to do it here as well to detect failures early.
         """
-        assert self.hydra_context is not None
+        config_loader = (
+            self.hydra_context.config_loader
+            if hasattr(self, "hydra_context") and self.hydra_context is not None
+            else self.config_loader  # type: ignore
+        )
+        assert config_loader is not None
+
```

Comment:
It's weird that an object needs to ask itself if it has a field.
I guess this is inherited from the current design, but I wonder if we should switch to simple setters from in the base class + a `setup()` that takes no parameters and uses the members already set by the setters.
What do you think?

Reply:
after offline chats, we've decided to keep it as it is. No changes in Plugins setup for now.